File: src/main/java/ru/sicampus/bootcamp2026/dto/MeetingDto.java
package ru.sicampus.bootcamp2026.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import ru.sicampus.bootcamp2026.enums.MeetingType;

import java.time.LocalDateTime;
import java.util.List;

@Data
public final class MeetingDto {
    private Long id;
    private @NotBlank String title;
    private @NotNull LocalDateTime startAt;
    private @NotNull LocalDateTime endAt;
    private String description;
    private MeetingType type;
    private String location;
    private String url;
    private List<String> inviteeLogins;
}

File: src/main/kotlin/ru/sicampus/bootcamp2026/dto/MeetingDto.kt
package ru.sicampus.bootcamp2026.dto

import java.time.LocalDate
import java.time.LocalDateTime
import java.time.LocalTime

data class MeetingResponseDto(
    val id: Long,
    val organizerId: Long,
    val title: String,
    val description: String?,
    val date: LocalDate,
    val startTime: LocalTime,
    val endTime: LocalTime,
    val createdAt: LocalDateTime
)

data class MeetingCreateDto(
    val organizerId: Long,
    val title: String,
    val description: String? = null,
    val date: LocalDate,
    val startTime: LocalTime,
    val endTime: LocalTime
)

data class MeetingUpdateDto(
    val title: String? = null,
    val description: String? = null,
    val date: LocalDate? = null,
    val startTime: LocalTime? = null,
    val endTime: LocalTime? = null
)
